server/auth.go

```diff
@@ -12,6 +16,9 @@ type Auth interface {
 type ClientAuth interface {
 	// Get options associated with a client
 	GetOpts() *clientOpts
+	// If TLS is enabled, return true and the TLS ConnectionState
+	// If not, return false
+	GetTLSConnectionState() (bool, tls.ConnectionState)
```

Comment:
Can you not just return a pointer, which can be nil if TLS is not enabled? Do we need to copy the struct via the return arg?